File: Algorithms/Warmup/Time Conversion/TimeConversion.java
import java.io.*;
import java.math.*;
import java.text.*;
import java.util.*;
import java.util.regex.*;

public class Solution {
    /*
     * Complete the timeConversion function below.
     */
    static String timeConversion(String s) {
        /*
         * Write your code here.
         */           
        int hour = Integer.parseInt(s.substring(0, 2)) % 12;
        if (s.endsWith("PM"))
            hour += 12;
        return String.format("%02d", hour) + s.substring(2, 8);
    }

    private static final Scanner scan = new Scanner(System.in);

    public static void main(String[] args) throws IOException {
        BufferedWriter bw = new BufferedWriter(new FileWriter(System.getenv("OUTPUT_PATH")));

        String s = scan.nextLine();

        String result = timeConversion(s);

        bw.write(result);
        bw.newLine();
        bw.close();
    }
}

File: Algorithms/Warmup/Time Conversion/TimeConversion.js
'use strict';

const fs = require('fs');

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => {
    inputString += inputStdin;
});

process.stdin.on('end', _ => {
    inputString = inputString.trim().split('\n').map(str => str.trim());

    main();
});

function readLine() {
    return inputString[currentLine++];
}

/*
 * Complete the timeConversion function below.
 */
function timeConversion(s) {
    /*
     * Write your code here.
     */
    let hours = parseInt(s.substr(0, 2));
    let mins = s.substr(3, 2);
    let secs = s.substr(6, 2);
    let pmOrAm = s.substr(-2, 2);
    
    if (pmOrAm === 'AM') {
        if (hours != 12) {
            return (s.substr(0, 2) + ":" + mins + ":" + secs);
        }
        else {
            return ("00:" + mins + ":" + secs);
        }        
    } else if (hours !== 12 && pmOrAm === 'PM') {
        return (hours+12+":"+mins+":"+secs);
    } else {
        return (hours+":"+mins+":"+secs);
    }  

}

function main() {
    const ws = fs.createWriteStream(process.env.OUTPUT_PATH);

    const s = readLine();

    let result = timeConversion(s);

    ws.write(result + "\n");

    ws.end();
}
